packages/ra-ui-materialui/src/detail/Edit.spec.tsx

```diff
@@ -106,6 +107,13 @@ describe('<Edit />', () => {
         });
     });
 
+    it("shoudln't display the record while loading", async () => {
+        render(<EmptyWhileLoading />);
+        expect(screen.queryByText('Book War and Peace')).toBeNull();
+        await new Promise(resolve => setTimeout(resolve, 3000));
+        expect(screen.queryByText('Book War and Peace')).not.toBeNull();
```

Comment:
This test doesn't make sense. Of course the record won't show up during loading, because the data isn't there. The purpose of the `emptyWhileLoading` prop is to avoid rendering the child altogether. 
So you should add a static string to the Edit child and make sure it's not rendered while the data provider is loading. 

Also, don't use timeouts. Check https://github.com/marmelab/react-admin/pull/10238/files#diff-1f5141788db3c747028c3cb2da9eaa6f9244e63d77addb0ea7eb70688fcb9ba9 for an example of test that uses a suspended Promise resolved bu the test itself.